lib/opentracing/reference.rb

```diff
@@ -63,9 +63,7 @@ def self.child_of(context)
     #   ])
     #
     def self.follows_from(context)
-      if context.is_a?(Span)
-        context = context.context
-      end
+      context = context.context if context.is_a?(Span)
```

Comment:
Why did u change the line in the PR, seems to be unrelated ? Notice that `self.child_of` contains exactly the same logic for SpanContext extraction.